### PR TITLE
Add OTEL Tracing

### PR DESCRIPTION
**Pull request by aacoba**
This MR Adds OpenTelemetry based tracing to Protect.
It uses [autoprop](https://pkg.go.dev/go.opentelemetry.io/contrib/propagators/autoprop) and [autoexport](https://pkg.go.dev/go.opentelemetry.io/contrib/exporters/autoexport).
This makes it configurable with the OTEL SDK configuration properties.
This maybe makes it a bit harder to configure as configuration is not done via protect.yml, but it does allow automatic instrumentation if the OTEL environment variables are set in the runtime environments.
If desired, I can also remove the autoconfiguration and make it all dependant on protect.yml, but this will extend the config by quite a bit and we will have to keep updating the config and code with newer options that we don't necessarily use for our own usecase.

### CODE REVIEW

**Review by ldebruijn — APPROVED** [review]

**Comment by ldebruijn** [comment]
Could you add documentation for configuring tracing?

File: internal/app/otel/otel.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	"errors"
+	"go.opentelemetry.io/contrib/exporters/autoexport"
+	"go.opentelemetry.io/contrib/propagators/autoprop"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.24.0"
+)
+
+func SetupOTelSDK(ctx context.Context, version string) (shutdown func(context.Context) error, err error) {
```
[the remark this answers is not in the document]
Comment (aacoba) [reply internal/app/otel/otel.go:14]: stolen from the otel getting started documentation, but I think in this case it makes it clear that it is a shutdown hook that you need to call to cleanly shutdown the tracer

**Comment by aacoba** [comment]
> Could you add documentation for configuring tracing?

Yes I will, the branch is still WIP

File: docs/run/tracing.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+
+```yaml
+...
+spec:
+  template:
+    spec:
+      containers:
+      - name: graphql-protect
+        image: eu.gcr.io/bolcom-stg-shop-api-a1c/graphql-protect:v0.11.0 # Replace with the appropriate version
```
Comment (ldebruijn) [inline docs/run/tracing.md:32]: we also tag with `latest`, perhaps a good idea to use for these examples

File: internal/http/proxy/proxy.go [hunk 3]
```diff
@@ -17,22 +16,26 @@ type Config struct {
 	Timeout   time.Duration `conf:"default:10s" yaml:"timeout"`
 	KeepAlive time.Duration `conf:"default:180s" yaml:"keep_alive"`
 	Host      string        `conf:"default:http://localhost:8081" yaml:"host"`
+	Tracing   TracingConfig `yaml:"tracing"`
+}
+
+type TracingConfig struct {
+	RedactedHeaders []string `yaml:"redactedheaders"`
```
Comment (ldebruijn) [inline internal/http/proxy/proxy.go:23]: Could you add this configuration option to your documentation please?

**Review by ldebruijn — CHANGES_REQUESTED** [review]
Thanks again for a great contribution! Just a few bits that could use improvement.

I do agree with your approach to configure OTEL via the environment variables, rather than configuring it via protect.yml 👍

File: internal/http/proxy/proxy.go (hunk 3 above)
Comment (ldebruijn) [reply internal/http/proxy/proxy.go:23]: also maybe update it to `redacted_headers` for readability

**Review by ldebruijn — APPROVED** [review]

**Pull request merged**